src/streaming/controllers/MediaController.js

```diff
@@ -171,7 +171,7 @@ function MediaController() {
 
         tracks[id][type].current = track;
 
-        if (tracks[id][type].current) {
+        if (tracks[id][type].current && track.isFragmented) {
```

Comment:
we need to check the type here as well. otherwise audio and video track switched will not work anymore

Reply:
fixed